File: py/1422.maximum-score-after-splitting-a-string.py
#
# @lc app=leetcode id=1422 lang=python3
#
# [1422] Maximum Score After Splitting a String
#

# @lc code=start
class Solution:
    def maxScore(self, s: str) -> int:
        one_count = s.count('1', 1)
        score = (1 if s[0] == '0' else 0) + one_count
        res = score
        for i in range(1, len(s)-1):
            if s[i] == '1':
                score -= 1
            else:
                score += 1
                res = max(res, score)
        return res
# @lc code=end

File: java/1422.maximum-score-after-splitting-a-string.java
/*
 * @lc app=leetcode id=1422 lang=java
 *
 * [1422] Maximum Score After Splitting a String
 */

// @lc code=start
class Solution {
    public int maxScore(String s) {
        int one_count = 0;
        int width = s.length();
        for (int i = 1; i != width; ++i)
        {
            if (s.charAt(i) == '1')
            {
                ++one_count;
            }
        }
        int score = (s.charAt(0) == '0' ? 1 : 0) + one_count;
        int res = score;

        for (int i = 1; i < width - 1; ++i)
        {
            if (s.charAt(i) == '1')
            {
                --score;
            }
            else
            {
                ++score;
                res = Math.max(res, score);
            }
        }
        return res;
    }
}
// @lc code=end

